roles/content_view_promotion_rollback_publish/tasks/promote.yml

```diff
@@ -0,0 +1,82 @@
+---
+# get data on the current content-view
+- name: "Gather Data For Current Content-View From Foreman"
+  theforeman.foreman.resource_info:
+    username: "{{ foreman.user }}"
+    password: "{{ foreman.password }}"
+    server_url: "{{ foreman.url }}"
+    organization: "{{ organization.key }}"
+    resource: content_views
+    search: name = "{{ content_view }}"
+    validate_certs: false
+  register: content_view_data
+
+
+# get data on the current content-view version
+- name: "Gather Data For Current Content-View Versions From Foreman"
+  theforman.foreman.resource_info:
+    username: "{{ foreman.user }}"
+    password: "{{ foreman.password }}"
+    server_url: "{{ foreman.url }}"
+    organization: "{{ organization.key }}"
+    resource: content_view_versions
+    params:
+      content_view_id: "{{ content_view_data.resources[0].id }}"
+  register: version_information
```

Comment:
There should be a dedicated module for this as well in the next few days: https://github.com/theforeman/foreman-ansible-modules/pull/1169

Could be good feedback before merging it whether it handles your use case.

Reply:
Please see comment above.